packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,1254 @@
+     * @param {number} rpcOptions.id ID used to identify request
+     * @param {string} rpcOptions.jsonrpc JSON RPC version
+     * @returns {Promise} Array of work info (in order: current block header pow-hash, seed hash used for the DAG, and boundary condition (“target”), 2^256 / difficulty)
+     */
+    async getWork(rpcOptions?: HttpRpcOptions): Promise<EthStringArrayResult> {
+        try {
+            return await this._requestManager.send({
+                ...rpcOptions,
+                method: 'eth_getWork',
+                jsonrpc: rpcOptions?.jsonrpc || this._DEFAULT_JSON_RPC_VERSION,
+                params: [],
+            });
+        } catch (error) {
+            throw Error(`Error getting work: ${error.message}`);
+        }
+    }
+
+    /**
+     * Used for submitting a proof-of-work solution
+     * @param {string} nonce Hex string representing found nonce (64 bits)
+     * @param {string} powHash Hex string representing POW hash (256 bits)
+     * @param {string} digest Hex string representing mix digest (256 bits)
+     * @param {object} rpcOptions RPC options
+     * @param {number} rpcOptions.id ID used to identify request
+     * @param {string} rpcOptions.jsonrpc JSON RPC version
+     * @returns {Promise} Returns true if the provided solution is valid, otherwise false
+     */
+    async submitWork(
+        nonce: string,
+        powHash: string,
+        digest: string,
+        rpcOptions?: HttpRpcOptions
+    ): Promise<EthBooleanResult> {
+        try {
+            return await this._requestManager.send({
+                ...rpcOptions,
+                method: 'eth_submitWork',
+                jsonrpc: rpcOptions?.jsonrpc || this._DEFAULT_JSON_RPC_VERSION,
+                params: [nonce, powHash, digest],
+            });
+        } catch (error) {
+            throw Error(`Error submitting work: ${error.message}`);
+        }
+    }
+
+    /**
+     * Used for submitting mining hashrate
+     * @param {string} hashRate Hex string representing desired hash rate (32 bytes)
+     * @param {string} clientId Hex string representing ID identifying the client
+     * @param {object} rpcOptions RPC options
+     * @param {number} rpcOptions.id ID used to identify request
+     * @param {string} rpcOptions.jsonrpc JSON RPC version
+     * @returns {Promise} Returns true if the provided solution is valid, otherwise false
+     */
+    async submitHashRate(
+        hashRate: string,
+        clientId: string,
+        rpcOptions?: HttpRpcOptions
+    ): Promise<EthBooleanResult> {
+        try {
+            return await this._requestManager.send({
+                ...rpcOptions,
+                method: 'eth_submitHashRate',
+                jsonrpc: rpcOptions?.jsonrpc || this._DEFAULT_JSON_RPC_VERSION,
+                params: [hashRate, clientId],
+            });
+        } catch (error) {
+            throw Error(`Error submitting hash rate: ${error.message}`);
+        }
+    }
+}
```

Comment:
It's a lot of code, but its also set it once, and forget it.

Generator would be nice, but as you even said having type safety (in a readable way) is better. Maybe we can revisit the generator that Nick showed us later!